syntax = "proto3";

service Identity {
   rpc createUpdateIdentity (hrData) returns (hrDataMessage) {}
   rpc readIdentity (readData) returns (identityData) {}
   rpc deleteIdentity (deleteData) returns (deleteMessage) {}
   rpc appearUserId (getUserId) returns (userId) {}
}

message readData {
    optional string hr_id = 1;
    optional string identity_id = 2;
}

message getUserId {
    string identity_id = 1;
    string first_name = 2;
    string middle_name = 3;
    string last_name = 4;
}

message userId {
    string identity_id = 1;
    string user_id = 2;
    string status_message = 3;
}

message hrData {
    string hr_id = 1;
    string legal_first_name = 2;
    string legal_middle_name = 3;
    string legal_last_name = 4;
    string preferred_first_name = 5;
    string preferred_middle_name = 6;
    string preferred_last_name = 7;
    string job_title_code = 8;
    string job_title_description = 9;
    string job_title_level = 10;
    string location_code = 11;
    string location_name = 12;
    string location_address = 13;
    string phone_number_work = 14;
    string phone_number_work_mobile = 15;
    string phone_number_home = 16;
    string phone_number_home_mobile = 17;
    string email_work_primary = 18;
    string email_home_primary = 19;
    string status = 20;
    string hire_start_date = 21;
    string termination_date = 22;
    string leave_of_absense_start_date = 23;
    string leave_of_absense_end_date = 24;
    string employee_type = 25;
    string cost_center = 26;
    string department_number = 27;
    string department_name = 28;
    string manager_hr_id = 29;
}

message hrDataMessage {
    string identity_id = 1;
    string hr_id = 2;
    string user_id = 3;
    string status_message = 4;
}

message identityData {
    string identity_id = 1;
    string create_ts = 2;
    string hr_id = 3;
    string user_id = 4;
    string legal_first_name = 5;
    string legal_middle_name = 6;
    string legal_last_name = 7;
    string preferred_first_name = 8;
    string preferred_middle_name = 9;
    string preferred_last_name = 10;
    string use_preferred_name = 11;
    string job_title_code = 12;
    string job_title_description = 13;
    string job_title_level = 14;
    string location_code = 15;
    string location_name = 16;
    string location_address = 17;
    string phone_number_work = 18;
    string phone_number_work_mobile = 19;
    string phone_number_home = 20;
    string phone_number_home_mobile = 21;
    string email_work_primary = 22;
    string email_home_primary = 23;
    string status = 24;
    string hire_start_date = 25;
    string termination_date = 26;
    string leave_of_absense_start_date = 27;
    string leave_of_absense_end_date = 28;
    string employee_type = 29;
    string cost_center = 30;
    string department_number = 31;
    string department_name = 32;
    string manager_identity_id = 33;
    string manager_hr_id = 34;
    string manager_user_id = 35;
}

message deleteData {
    optional string hr_id = 1;
    optional string identity_id = 2;
}

message deleteMessage {
    string status_message = 1;
}

// python -m grpc_tools.protoc --proto_path=. ./identity.proto --python_out=. --grpc_python_out=.